streams/src/test/java/org/apache/kafka/streams/integration/RestoreIntegrationTest.java

```diff
@@ -87,37 +85,34 @@
 import static org.apache.kafka.streams.integration.utils.IntegrationTestUtils.waitForStandbyCompletion;
 import static org.hamcrest.MatcherAssert.assertThat;
 import static org.hamcrest.core.IsEqual.equalTo;
-import static org.junit.Assert.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertTrue;
 
-@Category({IntegrationTest.class})
+@Timeout(600)
+@Tag("integration")
```

Comment:
I understand that this change is necessary for JUnit5 migration but in it's current state if we merge this PR, these tests will not execute. Here's why:

In `build.gradle`, these tests are marked for execution with JUnit5 and hence will enter the `else` block in
```
if (shouldUseJUnit5) {
      useJUnitPlatform {
        includeTags "integration"
      }
    } else {
      useJUnit {
        includeCategories 'org.apache.kafka.test.IntegrationTest'
      }
    }
```
where they will try to find the previous `@Category({IntegrationTest.class})` but will not find it and hence, these tests will not execute.

My suggestion would be to keep `@Category({IntegrationTest.class})` in these tests and make all other changes.
Once we have all tests migrated to Junit5, create a PR where we remove the Junit5 eligibility from `build.gradle` and at the same time, we replace this annotation with `@Tag("integration")`

Reply:
This is also a fair point, I will make the changes in the new commit.